pkg/request/pool.go

```diff
@@ -203,9 +202,20 @@ func (rp *Pool) Submit(request []byte) error {
 }
 
 // NextRequests returns the next requests to be batched.
-// It returns at most maxCount requests, and at most maxSizeBytes, in a newly allocated slice.
-// Return variable full indicates that the batch cannot be increased further by calling again with the same arguments.
-func (rp *Pool) NextRequests(ctx context.Context) [][]byte {
+func (rp *Pool) NextRequests() [][]byte {
+
+	if rp.isClosed() || rp.isStopped() {
+		rp.logger.Warnf("pool halted or closed, returning nil")
+		return nil
+	}
+
+	if !rp.isBatchingEnabled() {
+		rp.logger.Warnf("NextRequests is called when batching is not enabled")
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), rp.options.BatchTimeout)
```

Comment:
Is it not a problem that `Fetch` can return nil? we will return an empty batch no?

Reply:
What line are you referring to here?
Sometimes there are no requests so we will return an empty batch
Please can you clarify your question

Reply:
but how can we return an empty batch? we don't have empty blocks